test/tube_test.c

```diff
@@ -268,3 +268,39 @@ CTEST2(tube, print_tubes)
     ASSERT_TRUE(tube_manager_add(data->mgr, t, &data->err));
     tube_manager_print_tubes(data->mgr);
 }
+
+static int _mock_tube_walker(void *data,
+                             const spud_tube_id *tube_id,
+                             tube *t)
+{
+    UNUSED_PARAM(tube_id);
+    UNUSED_PARAM(t);
+    int *num_tubes = (int *) data;
+    (*num_tubes)++;
+    return 1;
```

Comment:
Really sorry, this is so close, but could you destroy the tube here, please?  Otherwise there's a leak.